### ISSUE TITLE
GPU memory consumption with cudnn.enabled = False

### ISSUE THREAD

**Issue by tlkvstepan** [pytorch/pytorch#4673]
```
import torch as th 
import torch.nn as nn
from torch.autograd import Variable
import subprocess
import torch.backends.cudnn as cudnn
cudnn.enabled = False

def get_gpu_memory_map():
    
    result = subprocess.check_output(
        [
            'nvidia-smi', '--query-gpu=memory.used',
            '--format=csv,nounits,noheader'
        ])

    gpu_memory = [int(x) for x in result.strip().split('\n')]
    gpu_memory_map = dict(zip(range(len(gpu_memory)), gpu_memory))
    
    return gpu_memory_map

def conv2D_block(in_channels, out_channels, kernel_size=3, stride=1):
    padding = kernel_size//2
    return nn.Sequential(nn.Conv2d(in_channels, out_channels, kernel_size=kernel_size, stride=stride, padding=padding),
                         nn.LeakyReLU(negative_slope=0.1, inplace=True),
                         nn.BatchNorm2d(out_channels))

downsample0 = conv2D_block(in_channels=64, out_channels=64, kernel_size=5, stride=2)
downsample0.float().cuda()
downsample0.eval()

inp = Variable(th.rand(1, 64, 3000/2, 3000/2).float().cuda(), volatile=True)
print 'before', get_gpu_memory_map()
embed0 = downsample0(inp);  th.cuda.empty_cache() 
print 'after downsample0', get_gpu_memory_map()   
```
According to this code, difference in used GPU memory  before and after convolution is  4809 MB - 1345 MB = 3.5 GB . This is too much since *inp* is just 536MB and *embed0* is 134 MB. From my experiments this issue is fixed if i delete nn.BatchNorm2d block.

What can be the reason for this strange behavior? 

PS: I turned off CUDNN to avoid another bug with ConvTranspose3d  [#4344]

Pyton2.7, PyTorch 0.3 (pip),  CUDA 8, CUDNN 7

**Comment by apaszke** [comment]
Convolutions need some extra scratch space, and our non-cuDNN CUDA convolutions aren't very efficient, but we don't plan to invest a lot of work into them, since cuDNN is usually going to be waaaaaay better.

Also, note that we're using GitHub issues for bug reports only, and all questions should be posted on [our forums](discuss.pytorch.org).

**Comment by tlkvstepan** [comment]
Thank you for reply! I will use forum for questions
